examples/data-sources/powerscale_snapshot_schedule/datasource.tf

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright (c) 2023 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://mozilla.org/MPL/2.0/
+
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+# PowerScale SMB shares provide clients network access to file system resources on the cluster
+
+# Returns a list of all the PowerScale Snapshot schedules
+data "powerscale_snapshot_schedule" "example_snapshot_schedule_all" {
+  filter {
+
+    # Used for query parameter, supported by PowerScale Platform API
+    #The direction of the sort.Supported Values:ASC , DESC
+    # dir = "ASC"
+    # Return no more than this many results at once (see resume).
+    # limit = 1
+    # Continue returning results from previous call using this token (token should come from the previous call, resume cannot be used with other options).
+    # resume = "resume-token"
+    # The field that will be used for sorting. Choices are id, name, path, pattern, schedule, duration, alias, next_run, and next_snapshot. Default is id.
+    # sort = "name"
+  }
+}
+output "powerscale_snapshot_schedule_all" {
+  value = data.powerscale_snapshot_schedule.example_snapshot_schedule_all
+}
+
+# Returns a list of PowerScale Snapshot schedules based on path filter block
+data "powerscale_snapshot_schedule" "example_snapshot_schedule" {
+  filter {
+    # Used to specify path of snapshot schedule
+    path = "/ifs/DirNew1"
+
+    # Used for query parameter, supported by PowerScale Platform API
+    #The direction of the sort.Supported Values:ASC , DESC
```

Comment:
I am not sure if the terraform user will care about these filters. It seems more useful to a client like a UI who would want sorting from the returned API. You might want to just create a filter on schedule name for the terraform user.

Reply:
Added names as filter.